cpp/src/arrow/engine/substrait/plan_internal.cc

```diff
@@ -43,7 +43,7 @@ Status AddExtensionSetToPlan(const ExtensionSet& ext_set, substrait::Plan* plan)
   auto uris = plan->mutable_extension_uris();
   uris->Reserve(static_cast<int>(ext_set.uris().size()));
   for (uint32_t anchor = 0; anchor < ext_set.uris().size(); ++anchor) {
-    auto uri = ext_set.uris()[anchor];
+    auto uri = ext_set.uris().at(anchor);
```

Comment:
`at` will raise an exception if the element doesn't exist - is that what you want? Especially given the `empty` check below

Reply:
As the `uris` is now an unordered_map, I think it will raise an error if we want to use the `[]` operator on a const unordered_map, and thus used the `.at()` implementation instead. Should I remove the `empty` check, or can there be any better approach?

Reply:
Hm...does this work?
```
util::string_view uri = ext_set.uris()[anchor];
```

Reply:
Nvm, per @jeroen 's comment below we should keep this as `at`.